lib/internal/streams/operators.js

```diff
@@ -147,6 +147,20 @@ async function * map(fn, options) {
   }
 }
 
+async function * filter(fn, options) {
```

Comment:
This is an async generator for error's sake (so that if `fn` is not a function it rejects like an async generator would)

I changed the original code a bunch to pass the tests I added)